pkg/util/validation/exporter/ring_test.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package exporter
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/kv/consul"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/dskit/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOverridesExporter_emptyRing(t *testing.T) {
+	ringStore, closer := consul.NewInMemoryClient(ring.GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	// Create an empty ring.
+	ctx := context.Background()
+	require.NoError(t, ringStore.CAS(ctx, ringKey, func(in interface{}) (out interface{}, retry bool, err error) {
+		return ring.NewDesc(), true, nil
+	}))
+
+	cfg := RingConfig{Enabled: true}
+	cfg.KVStore.Mock = ringStore
+
+	cfg.InstanceID = "instance-1"
+	cfg.InstanceAddr = "127.0.0.1"
+	i1, err := newRing(cfg, log.NewNopLogger(), nil)
+	require.NoError(t, err)
+	require.NoError(t, services.StartAndAwaitRunning(ctx, i1.client))
+	t.Cleanup(func() { require.NoError(t, services.StopAndAwaitTerminated(ctx, i1.client)) })
+
+	_, err = i1.isLeader()
+	require.ErrorIs(t, err, ring.ErrEmptyRing)
```

Comment:
Wouldn't be flaky? There's no guarantee this code runs before the lifecycler registered the instance in the ring.

Reply:
The lifecycler service is not started, see four lines above.